src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,183 @@
+import p5 from 'p5' // we need the p5.Color type to declare top-level colors
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# Entries (p5 Template)
+
+[<img
+  src="../../assets/img/Entries/1729.png"
+  width=700
+  style="margin-left: 1em; margin-right: 0.5em"
+/>](../../assets/img/Entries/1729.png)
+
+This visualizer allows you to step back and forth through a sequence, showing
+each entry as a written number.
+
+_This visualizer is meant to be used as a template for new visualizers based on
+the p5.js library. It includes explanatory comments and minimal examples of
+required and commonly used features._
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    // === visualizer name ===
+    // appears in the visualizer list and bundle card titles
+
+    static visualizationName = 'Entries (p5 Template)'
+
+    // === parameters ===
+    // top-level properties that the user can choose while creating the
+    // visualizer bundle
+
+    stepSize = 1 // the default value of this parameter is zero
+
+    /** md
+## Parameters
+    **/
+    params = {
+        /** md
+- **Step size:** How far to step when the user presses an arrow key.
+         **/
+        stepSize: {
+            value: this.stepSize,
+            forceType: 'integer',
+            displayName: 'Step size',
+            required: true,
+        },
+    }
+
+    // === internal properties ===
+    // top-level properties that are set and updated while the visualizer is
+    // running, beyond the user's direct control
+
+    // navigation state. this property will be initialized during setup, but
+    // TypeScript can't infer that, so we have to give a placeholder value
+    index = 0
+
+    // palette colors. these will also be initialized during setup, which is our
+    // first chance to create colors. for now, we use `undefined` as a
+    // placeholder value
+    bgColor: p5.Color | undefined = undefined
+    textColor: p5.Color | undefined = undefined
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the step size is positive
+        if (this.params.stepSize.value <= 0) {
+            status.isValid = false
+            status.errors.push('Step size must be positive')
+        }
+
+        return status
+    }
+
+    setup() {
+        // the base setup function includes the `createCanvas()` call that must
+        // appear in every p5 setup function
```

Comment:
Add "class" and capitalize and add punctuation: `// The base class setup() method includes ... in every p5 setup function.`

Reply:
To clarify: do you want all comments to be capitalized, including in-line comments?

Reply:
I would like utterances that are grammatically sentences to begin with capital letters and end with periods. :)

Reply:
Whoops, I didn't see this until after I reformatted the comments. I followed the convention that each "guidance comments" is capitalized (because a guidance comment is often many sentences) and each "model comments" is non-capitalized (even if it's a complete sentence; none are more than one sentence).

I also left periods off the ends of lines, as the kids are wont to do these days.

Feel free to revise and push the comment formatting.

Reply:
No that looked good! Turns out I don't mind tiny little short uncapitalized unpunctuated sentences, especially because here it serves that visual identification purpose of the three types of comments. (Although I probably wont write any that way myself!) Made a tiny PR because one "model comment" was capitalized, but it needed to be a "guidance comment" anyway (!) and I added a guidance comment for restarting the loop. Resolving.